core/src/Result.h

```diff
@@ -40,9 +41,43 @@ class Result
 public:
 	using time_point = std::chrono::steady_clock::time_point;
 
-	explicit Result(DecodeStatus status);
-	Result(std::wstring text, ByteArray rawBytes, std::vector<ResultPoint> resultPoints, BarcodeFormat format, time_point tt = std::chrono::steady_clock::now());
-	Result(std::wstring text, ByteArray rawBytes, int numBits, std::vector<ResultPoint> resultPoints, BarcodeFormat format, time_point tt = std::chrono::steady_clock::now());
+	explicit Result(DecodeStatus status) : _status(status) {}
+
+	Result(std::wstring&& text, ByteArray&& rawBytes, std::vector<ResultPoint>&& resultPoints, BarcodeFormat format)
+	    : _text(std::move(text)),
+	      _rawBytes(std::move(rawBytes)),
+		  _resultPoints(std::move(resultPoints)),
+	      _format(format)
+	{
+		_numBits = static_cast<int>(_rawBytes.size()) * 8;
+	}
+
+	Result(DecoderResult&& decodeResult, std::vector<ResultPoint>&& resultPoints, BarcodeFormat format)
```

Comment:
Can we move this one and the other above into .cpp, it the .h will be less messy?